rmw_dds_common/test/test_graph_cache.cpp

```diff
@@ -1127,3 +1128,259 @@ TEST(test_graph_cache, normal_usage)
+      "node_name",
+      "node_namespace",
+      identity_demangle,
+      identity_demangle,
+      &allocator,
+      &names_and_types);
+    EXPECT_EQ(ret, RMW_RET_NODE_NAME_NON_EXISTENT);
+    rcutils_reset_error();
+    rcutils_allocator_t zero_allocator = rcutils_get_zero_initialized_allocator();
+    ret = graph_cache.get_writer_names_and_types_by_node(
+      "node_name",
+      "node_namespace",
+      identity_demangle,
+      identity_demangle,
+      &zero_allocator,
+      &names_and_types);
+    EXPECT_EQ(ret, RMW_RET_INVALID_ARGUMENT);
+    rcutils_reset_error();
+  }
+
+  // Add one participant.
+  add_participants(graph_cache, {"participant1"});
+  {
+    rmw_topic_endpoint_info_array_t topic_endpoint_info_array_sub =
+      rmw_get_zero_initialized_topic_endpoint_info_array();
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_ret_t ret = graph_cache.get_writers_info_by_topic(
+      "topic1",
+      identity_demangle,
+      &allocator,
+      &topic_endpoint_info_array_sub);
+    EXPECT_EQ(ret, RMW_RET_OK);
+  }
+
+  // Add some nodes.
+  add_nodes(
+    graph_cache, {
+    {"participant1", "ns1", "node1"},
+    {"participant1", "ns1", "node2"},
+    {"participant1", "ns2", "node1"}});
+
+  add_entities(
+    graph_cache,
+  {
+    // topic1
+    {"reader1", "topic1", "Str", true},
+    {"writer1", "topic1", "Str", false},
+  });
+
+  // Associate entities
+  associate_entities(
+    graph_cache,
+  {
+    // participant1, ns1, node1
+    {"reader1", true, "participant1", "ns1", "node1"},
+    {"writer1", false, "participant1", "ns1", "node2"},
+  });
+  {
+    rmw_topic_endpoint_info_array_t topic_endpoint_info_array_sub =
+      rmw_get_zero_initialized_topic_endpoint_info_array();
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_ret_t ret = graph_cache.get_writers_info_by_topic(
+      "topic1",
+      identity_demangle,
+      &allocator,
+      &topic_endpoint_info_array_sub);
+    EXPECT_EQ(ret, RMW_RET_OK);
+    rcutils_reset_error();
+  }
+
+  {
+    rmw_topic_endpoint_info_array_t topic_endpoint_info_array_sub =
+      rmw_get_zero_initialized_topic_endpoint_info_array();
+    rcutils_allocator_t failing_allocator = get_failing_allocator();
+    rmw_ret_t ret = graph_cache.get_writers_info_by_topic(
+      "topic1",
+      identity_demangle,
+      &failing_allocator,
+      &topic_endpoint_info_array_sub);
+    EXPECT_EQ(ret, RMW_RET_BAD_ALLOC);
+    rcutils_reset_error();
+  }
+  {
+    rmw_topic_endpoint_info_array_t topic_endpoint_info_array_sub =
+      rmw_get_zero_initialized_topic_endpoint_info_array();
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_ret_t ret = graph_cache.get_readers_info_by_topic(
+      "topic1",
+      identity_demangle,
+      &allocator,
+      &topic_endpoint_info_array_sub);
+    EXPECT_EQ(ret, RMW_RET_OK);
+    rcutils_reset_error();
```

Comment:
As there is a failed allocator test for `get_writers_info_by_topic` sounds appropriate to include as well a failing allocator one for `get_readers_info_by_topic` as well

Reply:
https://github.com/ros2/rmw_dds_common/pull/14/commits/9f4a0f5269cc7a5e6e85f1356d236d00cd60c5c9